BetterShardsBukkit/src/main/java/vg/civcraft/mc/bettershards/manager/ConnectionManager.java

```diff
@@ -79,17 +86,48 @@ public boolean teleportPlayerToServer(Player p, String server, PlayerChangeServe
 			BetterShardsPlugin.getInstance().getLogger().log(Level.INFO, "During BetterShards teleport, removing player {0} from vehicle", p.getUniqueId());
 			p.getVehicle().eject();
 		}
-		if (p.isDead()) {
+		if (p.isDead() && !delayedTransit.contains(p.getUniqueId())) {
 			throw new PlayerStillDeadException();
 		}
 		transitManager.addPlayerToExitTransit(p.getUniqueId(), server); // So the player isn't tried to be sent twice.
 		MercuryManager.warnOfArrival(p.getUniqueId(), server); //so target server prepares
 		CustomWorldNBTStorage st = CustomWorldNBTStorage.getWorldNBTStorage();
 		st.save(p, st.getInvIdentifier(p.getUniqueId()), true);
+		
+		p.setGameMode(GameMode.SPECTATOR); // So they cant do anything.
```

Comment:
This is what dan recommended so @ProgrammerDan.
I believe it gets the lock before hand if done async and then saves out and makes the login wait